e2e/testcases/override_resource_limits_test.go

```diff
@@ -234,26 +197,39 @@ func TestOverrideReconcilerResourcesV1Alpha1(t *testing.T) {
 	// Verify the resource requests and limits of root-reconciler are not affected by the resource changes of ns-reconciler-backend and ns-reconciler-fronend
 	err = nt.Validate(rootReconcilerNN.Name, rootReconcilerNN.Namespace, &appsv1.Deployment{},
 		testpredicates.GenerationEquals(rootReconcilerDeploymentGeneration),
-		testpredicates.HasCorrectResourceRequestsLimits(reconcilermanager.Reconciler, resource.MustParse("500m"), resource.MustParse("800m"), resource.MustParse("400Mi"), resource.MustParse("411Mi")),
-		testpredicates.HasCorrectResourceRequestsLimits(reconcilermanager.GitSync, defaultGitSyncCPURequest, defaultGitSyncCPULimits, defaultGitSyncMemRequest, defaultGitSyncMemLimits))
+		testpredicates.DeploymentContainerResourcesEqual(updatedRootReconcilerResources),
+		testpredicates.DeploymentContainerResourcesEqual(defaultResources[reconcilermanager.GitSync]),
+	)
 	if err != nil {
 		nt.T.Fatal(err)
 	}
 
+	updatedBackendReconcilerResources := v1beta1.ContainerResourcesSpec{}
```

Comment:
Is this snippet just testing schema conversion? Curious if that needs to be an e2e test or if it could be a unit test

Reply:
The default map function returns v1beta1 objects, so we have to convert them to alpha to do the validation in this test.

We could make a function to return v1alpha1 defaults, but I didn't want to make the binary depend on the conversion code. So far, we're only using the conversion code in tests.

Reply:
added helper